docs/syntax/syntax.md

```diff
@@ -205,95 +205,429 @@ Is below, but it won't be parsed into the document.
 
 (syntax/referencing)=
 
-## Markdown Links and Referencing
+## Links and Referencing
 
-Markdown links are of the form: `[text](link)`.
+CommonMark links come in three forms ([see the spec](https://spec.commonmark.org/0.30/#links)):
 
-If you set the configuration `myst_all_links_external = True` (`False` by default),
-then all links will be treated simply as "external" links.
-For example, in HTML outputs, `[text](link)` will be rendered as `<a href="link">text</a>`.
+*Autolinks* are [URIs][uri] surrounded by `<` and `>`:
 
-Otherwise, links will only be treated as "external" links if they are prefixed with a scheme,
-configured with `myst_url_schemes` (by default, `http`, `https`, `ftp`, or `mailto`).
-For example, `[example.com](https://example.com)` becomes [example.com](https://example.com).
+```md
+<scheme:path?query#fragment>
+```
 
-:::{note}
-The `text` will be parsed as nested Markdown, for example `[here's some *emphasised text*](https://example.com)` will be parsed as [here's some *emphasised text*](https://example.com).
+*Inline links* allow for optional explicit text and titles (in HTML titles are rendered as tooltips):
+
+```md
+[Explicit *Markdown* text](destination "optional explicit title")
+```
+
+or, if the destination contains spaces,
+
+```md
+[text](<a destination>)
+```
+
+*Reference links* define the destination separately in the document, and can be used multiple times:
+
+```md
+[Explicit *Markdown* text][label]
+
+[label]: destination "optional explicit title"
+```
+
+MyST, supports the following destination types:
+
+|     Link Type      |            Auto            |          Inline           | Single Page[^sp] |
```

Comment:
I wonder if it also might be helpful to split this out into a "spec" which could be our first enhancement proposal, which could go into all of these details, and the user facing docs.

In the user facing docs, I don't think there is ever really a use case where we should encourage people to use the `project:` form, as that is only necessary to suffice the autolink requiring a protocol.

That is, some of these things are important to discuss from a spec side, but not that useful to encourage or even show to users (or in the worst case, they just confuse people!).

Reply:
I agree - I am coming at this from a user's perspective, not a spec perspective. I think for the user docs we should choose the 1-3 most important workflows that are simplest and cover 80% of use-cases, and have them dig deeper if they really want to know all the things that MyST could do.

Reply:
> In the user facing docs, I don't think there is ever really a use case where we should encourage people to use the project: form, as that is only necessary to suffice the autolink requiring a protocol.

I wouldn't say this is necessarily the case, because if you are not using explicit text, then e.g. `[](file.md)` just "dissapears" in a commonmark renderer (it is here: [](file.md)), whereas for `<project:file.md>` you can at least actually see it rendered: <project:file.md>.
In general I feel it may be better to use auto-links, for things with implicit text

Reply:
I feel that all the things I've put in this new "Links and Referencing" section are all things that I use all the time as a user, and wish I'd known more easily how to do.
I all see lots of other people asking how to do them in the forums.

I guess I'd ask; what part of this documentation do you feel should not be there?

Reply:
For example, before I wrote this, it was really unclear how sphinx numbering worked, and how to work out what to reference from an external project.

Just having the `myst-inv` cmdline tool and `sphinx-build -b myst_refs` builder, I feel are going to be so helpful, in working out what references are in your project.